tdee_macro_calc/main.py

```diff
@@ -1,81 +1,56 @@
-    # Target daily calorie deficit (3.2kcal/lb)
-    # Target daily calorie intake (Current TDEE minus target daily calorie deficit)
-    # Macros - first hit the calorie goal, then protein, then fat, then fill rest with carbs
-        # Protein 4 cal per gram, .8-1.0g/lb for building muscle, 120g-160g/day is a good place to start
-            # Output: how many grams of protein you should eat (and as an FYI, how many calories that is)
-        # Fat 9 cal per gram, minimum .3g/lb
-            # Output: how many grams of fat you should eat (and as an FYI, how many calories that is)
-        # Carbs 4 cal per gram Essentially, take the calories eaten in protein and fat, subtract from your goal calories per day, and the rest is what you can eat in carbs. However, carbs/fat really doesn't matter, what is most important is protein.
-            # Output: how many grams of carbs you should eat (and as an FYI, how many calories that is)
-
-# JSON Data structure DRAFT
-data = {
-    "user": {
-        "current_weight": 176, # this will be updated with a rolling 7-day average
-        "current_tdee": 2400, # this will be updated with a max 4 week (?) average
-        "sex": "male", # this determines default of 3.2, if female can go to 4.5 I believe. Not sure if I want to actually include this as functionality right now or not
-        "target_calorie_intake": 1700,
-        "per_lb_calorie_deficit": 3.2, # use 3.2 as default but allow user to change, possibly. TODO look at study to confirm dropping this number works
-        "per_lb_protein": .8, # use .8 as default but allow user to change, and/or play with different numbers in memory
-        "per_lb_fat": .3 # use .3 as default but allow user to change, and/or play with different numbers in memory
-    },
-    "tdee": [ # TODO This could totally be an object, same with user
-        {
-            "date": "9/20/22", # TODO this should be a datetime object and need to handle type checking
-            "weight": 178.1,
-            "calories": 2531
-        },
-        {
-            "date": "9/21/22",
-            "weight": 177.3,
-            "calories": 1971
-        },
-        {
-            "date": "9/22/22",
-            "weight": 179.2,
-            "calories": 2391
-        }
-    ]
-}
-
-def add_to_file():
+def check_data_file() -> dict:
+    """
+    Checks to see if the file data.json exists, and if not it creates it with default values.
+
+    Returns a dictionary of the user and tdee information in the file.
+    """
+    # Gather current user's directory information
+    home = str(Path.home())
+    tdee_directory = os.path.join(home, ".tdee")
+    
+    # If the .tdee directory doesn't exist in the user's home folder, create it.
+    if not os.path.isdir(tdee_directory):
+        try:
+            os.mkdir(tdee_directory)
+        except:
+            print("Directory already exists or another error.")
+
+    tdee_data = os.path.join(tdee_directory, "data.json")
```

Comment:
This all looks good, but I wanted to propose that we take advantage of some of the syntax sugar in Python's pathlib library since we're already importing it here.

We could do something like this

```python
# Get home dir.
home = Path.home()
# Build our data dir.
tdee_directory = home / ".tdee"
tdee_directory.mkdir(exists_ok=True)
tdee_file = tdee_directory / "data.json"
```

Then, we can write to this file using
```python
tdee_file.write_text(json.dumps(...))
```

Reply:
What is this sorcery!? Note to self, read the documentation for libraries I import 😬 This is so awesome, I'm learning so much.

That is cool, I'll play around with it and work it into the code!

Reply:
Quick question, the ellipses you used in this line
`tdee_file.write_text(json.dumps(...))`
was that meant to be an ellipses operator or just be a placeholder for future code to be filled in? If it's an ellipses operator I'm having a hard time deciphering it despite searching StackOverflow.

Reply:
Oops! No I didn't mean the operator—just a a placeholder. Sorry for the confusion!